qiime2/util.py

```diff
@@ -114,3 +115,32 @@ def duplicate(src, dst):
             shutil.copyfile(src, dst)
         else:
             raise
+
+
+def get_filepath_from_package(package, relative_filepath):
+    """Utility function for getting the absolute path to a packaged data asset.
+
+    Parameters
+    ----------
+    package : str
+        The name of the package or sub-package where the file can be found.
+    relative_filepath : str
+        The filepath to look up, relative to `package`.
+
+    Returns
+    -------
+    fp : pathlib.Path
+        The absolute filepath to the requested data asset.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the filepath does not exist.
+
+    """
+    fp = importlib.resources.files(package) / relative_filepath
+    if not fp.exists():
+        raise FileNotFoundError(
+            f'The requested data asset {fp} does not exist.')
+    else:
+        return fp
```

Comment:
Is there a reason you didn't include the `result_as_str` param in this implementation? Seems like it would be convenient since you're returning all of the test cases above as str(fp).

Reply:
@lizgehret, i was doing that just to maintain the API of the TestPluginBase.get_data_path function, which I suspect is the only one of these methods that is being called from outside of the framework. 

My goal is that users of this new function get a pathlib.FilePath object back since I think that's the type that makes the most sense (hence no option for a string), but for all of the places where TestPluginBase.get_data_path is already used, it makes sense to have that continue returning a string so we don't break other code. 

Let me know if you want to discuss more.

Reply:
Gotcha, that makes sense!